modules/oe_content_call_proposals/src/CallForProposalsNodeWrapper.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_content_call_proposals;
+
+use Drupal\Component\Render\MarkupInterface;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\oe_content\EntityWrapperBase;
+
+/**
+ * Wrap the "Call for proposals" content type by adding business methods.
+ *
+ * @method static CallForProposalsNodeWrapperInterface getInstance(\Drupal\Core\Entity\ContentEntityInterface $entity)
+ */
+class CallForProposalsNodeWrapper extends EntityWrapperBase implements CallForProposalsNodeWrapperInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getEntityId(): string {
+    return 'node';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getEntityBundle(): string {
+    return 'oe_call_proposals';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function hasOpeningDate(): bool {
+    return !$this->entity->get('oe_call_proposals_opening_date')->isEmpty();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function hasDeadlineDate(): bool {
+    return !$this->entity->get('oe_call_proposals_deadline')->isEmpty();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getOpeningDate(): ?DrupalDateTime {
+    if (!$this->hasOpeningDate()) {
+      return NULL;
+    }
+    $opening_date = $this->entity->get('oe_call_proposals_opening_date')->date;
+    // Prevent upcoming status when now & opening dates are the same.
+    $opening_date->setTime(0, 0, 0);
+    return $opening_date;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDeadlineDate(): ?DrupalDateTime {
+    if (!$this->hasDeadlineDate()) {
+      return NULL;
+    }
+    return $this->entity->get('oe_call_proposals_deadline')->date;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getStatus(): string {
```

Comment:
I think `public function getModel(): string {},` and `public function getModelLabel` can be also useful. If so then might be we should define
```
public static getModelsList() {
return [
    CallForProposalsNodeWrapperInterface::MODEL_SINGLE_STAGE => 'Single-stage',
    CallForProposalsNodeWrapperInterface::MODEL_TWO_STAGE => 'Two-stage',
    CallForProposalsNodeWrapperInterface::MODEL_MULTIPLE_CUT_OFF => 'Multiple cut-off',
    CallForProposalsNodeWrapperInterface::MODEL_PERMANENT => 'Permanent',
  ]
}
```
that we could use inside `oe_content_call_proposals_model_allowed_values()`.

With this approach all labels and keys of models will be in one place - CallForProposalsNodeWrapper

Reply:
I refactored the related functionality.